src/openrct2/ride/TrackDesign.cpp

```diff
@@ -351,13 +348,12 @@ rct_string_id TrackDesign::CreateTrackDesignMaze(const Ride& ride)
         return STR_TRACK_TOO_LARGE_OR_TOO_MUCH_SCENERY;
     }
 
-    gTrackPreviewOrigin = { static_cast<int16_t>(startLoc.x), static_cast<int16_t>(startLoc.y),
-                            (int16_t)(startLoc.element->base_height * 8) };
+    gTrackPreviewOrigin = { startLoc.x, startLoc.y, startLoc.element->base_height * 8 };
 
     // x is defined here as we can start the search
     // on tile start_x, start_y but then the next row
     // must restart on 0
-    for (int16_t y = startLoc.y, x = startLoc.y; y < 8192; y += 32)
+    for (int32_t y = startLoc.y, x = startLoc.y; y < 8192; y += 32)
```

Comment:
Magic value for maximum map size * 32.

We should have `constexpr`s for both this value and for 32, I think.

This may be picked up in another PR if desired.

Reply:
I'll pick it up on a new PR